test/integration/weighted_cluster_integration_test.cc

```diff
@@ -0,0 +1,209 @@
+#include <cstdint>
+#include <iterator>
+#include <numeric>
+
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/route/v3/route_components.pb.h"
+#include "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.pb.h"
+
+#include "test/integration/filters/repick_cluster_filter.h"
+#include "test/integration/http_integration.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+// For the simplicity of testing purpose, the pattern of the cluster array is below:
+// [0, ClusterNameClusterCount) will be clusters with `name` field specified.
+// [ClusterNameClusterCount, ClusterCount) will be clusters with `cluster_header` field specified.
+constexpr int ClusterCount = Envoy::RepickClusterFilter::ClusterCount;
+constexpr int ClusterNameClusterCount = Envoy::RepickClusterFilter::ClusterNameClusterCount;
+
+class WeightedClusterIntegrationTest : public testing::Test, public HttpIntegrationTest {
+public:
+  WeightedClusterIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, Network::Address::IpVersion::v6) {
+    default_weights_.reserve(ClusterCount);
+    std::fill_n(std::back_inserter(default_weights_), ClusterNameClusterCount, 20);
+    std::fill_n(std::back_inserter(default_weights_),
+                Envoy::RepickClusterFilter::ClusterHeaderClusterCount, 30);
+  }
+
+  void createUpstreams() override {
```

Comment:
Looking at it more, you cam remove the `createUpstreams()` and just call `setUpstreamCount` in the constructor.

Reply:
The framework will create upstreams for you. This function just duplicates the code in the base class.

Reply:
I took a closer look at this and tried it in several places. I don't think just `setUpstreamCount` will work: 
`createUpstreams()` here will invoke `addFakeUpstream` which will call this version of `FakeUpstream` constructor `FakeUpstream(Network::Test::createRawBufferSocketFactory(), makeListenSocket(config, makeAddress(port, version)), config)` via  `std::make_unique<FakeUpstream>(0, version_, config)`

However, the  `createUpstreams()` in the framework which is `BaseIntegrationTest::createUpstreams()` will invoke different constructor of `FakeUpstream(std::move(transport_socket_factory), makeListenSocket(config, address), config)` via `new FakeUpstream(std::move(factory), endpoint, upstreamConfig()`. 

I didn't investigate further because `createUpstreams()` has been used in many integration tests (i.e. I feel such change doesn't introduce extra complexity) but i feel the way of setting of the ListenSocket address (mentioned above) is the key here.  Also, I do feel the framework might need some clean-up/refactor to avoid confusion like this(which can be a TODO).